pkg/services/storage/utils.go

```diff
@@ -30,3 +33,23 @@ func buildGoTemplate(
 	}
 	return goTemplateParams
 }
+
+func createBlobContainer(storageAccountName, accessKey, containerName string) error { // nolint: lll
+	client, _ := storage.NewBasicClient(storageAccountName, accessKey)
+	blobCli := client.GetBlobService()
+	container := blobCli.GetContainerReference(containerName)
+	options := storage.CreateContainerOptions{
+		Access: storage.ContainerAccessTypePrivate,
+	}
+	_, err := container.CreateIfNotExists(&options)
```

Comment:
As it is a child service instead of "allInOne", the name could be conflictive among child instances. I think it should be not use `Create"IfNotExists"`.

Reply:
And I think fixing the method using here is not enough -- if the provision failed here, triggering deprovision, the existing container would be deleted.

Reply:
Agree with you. It is tricky, thanks for the reminding.